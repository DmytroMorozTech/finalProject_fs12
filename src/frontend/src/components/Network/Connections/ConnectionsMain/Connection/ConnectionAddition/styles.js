import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  menuItem: {
    '&:hover': {
      backgroundColor: 'rgba(0, 0, 0, 0.08)'
    }
  },

  item: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    fontSize: theme.typography.h5.fontSize,
    fontWeight: theme.typography.fontWeightBold,
    color: theme.palette.grey[700],
    cursor: 'pointer'
  },

  icon: {
    display: 'flex'
  },

  titles: {
    paddingLeft: theme.spacing(2)
  }
}))
