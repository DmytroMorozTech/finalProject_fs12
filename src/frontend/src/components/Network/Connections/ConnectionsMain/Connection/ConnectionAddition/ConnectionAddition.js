import { MenuItem } from '@material-ui/core'
import React from 'react'
import styles from './styles'
import DeleteIcon from '@material-ui/icons/Delete'

function ConnectionAddition (props) {
  const classes = styles()

  const {onClick} = props

  return (
    <div>
      <MenuItem className={classes.menuItem}>
        <div className={classes.item} onClick={onClick}>
          <div className={classes.icons}>
            <DeleteIcon fontSize='inherit'/>
          </div>
          <div className={classes.titles}>Remove connection</div>
        </div>
      </MenuItem>
    </div>
  )
}

export default ConnectionAddition
