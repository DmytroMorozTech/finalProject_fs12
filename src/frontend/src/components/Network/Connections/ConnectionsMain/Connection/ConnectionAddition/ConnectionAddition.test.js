import React from 'react'
import ConnectionAddition from './ConnectionAddition'
import { shallow } from 'enzyme'

const setUp = (props) => shallow(<ConnectionAddition {...props}/>)

describe('should render ConnectionAddition component', () => {
  let component
  let onClick = () => {}
  beforeEach(() => {
    component = setUp({onClick})
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
