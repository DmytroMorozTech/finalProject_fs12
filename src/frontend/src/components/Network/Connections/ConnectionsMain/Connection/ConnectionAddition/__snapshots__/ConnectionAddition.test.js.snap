// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConnectionAddition component to match snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(MenuItem))
    className="makeStyles-menuItem-1"
  >
    <div
      className="makeStyles-item-2"
      onClick={[Function]}
    >
      <div
        className="makeStyles-icon-3"
      >
        <Memo(ForwardRef(DeleteIcon))
          fontSize="large"
        />
      </div>
      <div
        className="makeStyles-titles-4"
      >
        Remove connection
      </div>
    </div>
  </WithStyles(ForwardRef(MenuItem))>
</div>
`;
